docs/src/man/split_apply_combine.md

```diff
@@ -778,3 +893,339 @@ julia> df
+in an array. This similarity of functionality was the reason why this name was
+picked:
+
+```jldoctest sac
+julia> collect(eachindex(df.customer_id))
+6-element Vector{Int64}:
+ 1
+ 2
+ 3
+ 4
+ 5
+ 6
+```
+
+This, for example, means that in the following example the two created columns
+have the same contents:
+
+```jldoctest sac
+julia> combine(gdf, eachindex, :customer_id => eachindex)
+6×3 DataFrame
+ Row │ customer_id  eachindex  customer_id_eachindex
+     │ String       Int64      Int64
+─────┼───────────────────────────────────────────────
+   1 │ a                    1                      1
+   2 │ b                    1                      1
+   3 │ b                    2                      2
+   4 │ b                    3                      3
+   5 │ c                    1                      1
+   6 │ c                    2                      2
+```
+
+
+## Column-independent operations versus functions
+
+When discussing column-independent operations it is important to remember
+that operation specification syntax allows you to pass a function (without
+source and target column names), in which case such a function gets passed a
+`SubDataFrame` that represents a group in a `GroupedDataFrame`. Here is an
+example comparing column-independent operation and a function:
+
+```jldoctest sac
+julia> combine(gdf, nrow, x -> nrow(x))
+3×3 DataFrame
+ Row │ customer_id  nrow   x1
+     │ String       Int64  Int64
+─────┼───────────────────────────
+   1 │ a                1      1
+   2 │ b                3      3
+   3 │ c                2      2
+```
+
+Notice that columns `:nrow` and `:x1` have an identical contents. This is
+expected. We already know that `nrow` is a column-independent operation
+generating the `:nrow` column with number of rows per group. However, the
+`x -> nrow(x)` anonymous function does exactly the same as it gets a
+`SubDataFrame` as its argument and returns its number of rows (the `:x1` column
+name is a default auto-generated column name in this case).
+
+Passing a function taking a `SubDataFrame` is a flexible functionality allowing
+you to perform complex operations on your data. However, you should bear in mind
+two aspects:
+
+* Using the full operation specification syntax (where source and target column
+  names are passed) will lead to faster execution of your code (as the Julia
+  compiler is able to better optimize execution of such operations) in
+  comparison to just passing a function taking a `SubDataFrame`.
+* Although writing `row`, `proprow`, `groupindices`, and `eachindex` looks like
+  just passing a function they **do not** take a `SubDataFrame` as their
+  argument. As we explained in this section, they are special column-independent
+  operations that are exceptions to the standard operation specification syntax
+  rules. They were added for user convenience (and at the same time they are
+  optimized to be fast).
```

Comment:
This point is a bit weird given that the example above using `nrow` shows that the result is exactly the same. Maybe explain that it's the case for `nrow` and `eachindex` (though they are optimized) but not for `proprow` and `groupindices`?

Reply:
I agree and this competition/interference was the most difficult part of learning column-independent expressions in DF.jl for me (so I'm happy to see it get an explanation in the docs!).

To that point, maybe a slight change in wording? Perhaps instead of:
> they **do not** take a `SubDataFrame` as their argument

It could be re-written as something like this, with emphasis on "routine case"?:
> they **do not** follow the routine case of a function taking a `SubDataFrame` as their argument. {Explain internal optimization in the case of `nrow` and `eachindex`}. {Explain that `proprow` and `groupindices` actually take different arguments}.

For what it's worth data.table also does [internal optimization of expressions which they call GForce](https://rdatatable.gitlab.io/data.table/reference/datatable-optimize.html) (including `.N`, although that's not a stand-alone function in data.table). I wonder if there's some good language in that vignette which can be borrowed for this.

Reply:
> This point is a bit weird given that the example above using `nrow` shows that the result is exactly the same.

The result above is not the same - it has a different column name. Actually the case that it would work is only for `nrow`. For `eachindex` it is also an error as for `proprow` and `groupindices`. I have added an example showing the difference for `eachindex`.

I have updated the description an examples. I hope they are clearer now.

Regarding GForce - this is an independent issue. We have GForce-like optimizations also in other places and introducing it would risk confusing them. I just removed the reference to optimizations (although they are made indeed).